utils/funcexp/func_round.cpp

```diff
@@ -200,6 +203,67 @@ double Func_round::getDoubleVal(Row& row,
     return d;
 }
 
+long double Func_round::getLongDoubleVal(Row& row,
+                                FunctionParm& parm,
+                                bool& isNull,
+                                CalpontSystemCatalog::ColType& op_ct)
+{
+    if (execplan::CalpontSystemCatalog::LONGDOUBLE == op_ct.colDataType)
+    {
+        int64_t d = 0;
+        long double  p = 1;
+
+        if (parm.size() > 1)  // round(X, D)
+            decimalPlaceDouble(parm, d, p, row, isNull);
+
+        if (isNull)
+            return 0.0;
+
+        long double x = parm[0]->data()->getLongDoubleVal(row, isNull);
+
+        if (!isNull)
+        {
+            x *= p;
+
+            if (x >= 0)
+                x = floor(x + 0.5);
+            else
+                x = ceil(x - 0.5);
+
+            if (p != 0.0)
+                x /= p;
+            else
+                x = 0.0;
+        }
+
+        return x;
+    }
+
+    if (isUnsigned(op_ct.colDataType))
+    {
+        return getUintVal(row, parm, isNull, op_ct);
+    }
+
+    IDB_Decimal x = getDecimalVal(row, parm, isNull, op_ct);
+
+    if (isNull)
+        return 0.0;
+
+    double d = x.value;
+
+    if (x.scale > 0)
+    {
+        while (x.scale-- > 0)
+            d /= 10.0;
+    }
+    else
+    {
+        while (x.scale++ < 0)
+            d *= 10.0;
+    }
```

Comment:
I would have thought using exp10() (or similar) would be more optimal to the above.

Reply:
Probably so. This code was already there and I didn't see the improvement possibility.